types.go

```diff
@@ -0,0 +1,37 @@
+package config
+
+import (
+	"encoding/json"
+)
+
+// Strings is a helper type that (un)marshals a single string to/from a single
+// JSON string and a slice of strings to/from a JSON array of strings.
+type Strings []string
+
+// UnmarshalJSON conforms to the json.Unmarshaler interface.
+func (o *Strings) UnmarshalJSON(data []byte) error {
+	if data[0] == '[' {
+		return json.Unmarshal(data, (*[]string)(o))
+	}
+	var value string
+	if err := json.Unmarshal(data, &value); err != nil {
+		return err
+	}
+	*o = []string{value}
+	return nil
+}
+
+// MarshalJSON conforms to the json.Marshaler interface.
+func (o Strings) MarshalJSON() ([]byte, error) {
+	switch len(o) {
+	case 0:
+		return json.Marshal(nil)
```

Comment:
Shouldn't this be returning an empty array? Or will the marshaller just omit the entry if it's nil?

Reply:
Exactly. Basically, I'd like to use `null` for the case where *no* API addresses are specified.

Reply:
@overbool want to file a PR against go-ipfs that uses this?

Reply:
@Stebalien Did you mean this https://github.com/ipfs/go-ipfs/issues/5401?